### PR TITLE
Conversion fixes

### PR DESCRIPTION
**Pull request by fredrikekre**
Some fixes
- Conversion between general 4th order data
- Conversion between data types (Int -> Float)
- ~~Conversion from "full data" to SymmetricTensor, if the data is symmetric~~
- Create a `SymmetricTensor` from symmetric general data

### CODE REVIEW

File: src/promotion_conversion.jl [hunk 1]
```diff
@@ -113,15 +115,29 @@ end
...
+@generated function Base.convert{order, dim, T}(Tt::Type{Tensor{order, dim, T}}, data)
+    N = n_components(Tensor{order, dim})
     return quote
         @assert length(data) == $N
         Tv = promote_type(T, eltype(data))
         get_main_type(Tt){order,dim,Tv, $N}(to_tuple(NTuple{$N, Tv}, data))
     end
 end
 
+@generated function Base.convert{order, dim, T}(Tt::Type{SymmetricTensor{order, dim, T}}, data)
```
Comment (KristofferC) [inline src/promotion_conversion.jl]: This should in my opinon not be in `convert` but in the constructor like:

``` jl
julia> SymmetricTensor{2,2}([1.0,2.0,3.0])
2×2 ContMechTensors.SymmetricTensor{2,2,Float64,3}:
 1.0  2.0
 2.0  3.0
```

**Comment by fredrikekre** [comment]
Yea, that makes more sense ... Just nice to be able to do this:

``` jl
julia> SymmetricTensor{2,2}([1.0, 2.0, 2.0, 3.0])
2×2 ContMechTensors.SymmetricTensor{2,2,Float64,3}:
 1.0  2.0
 2.0  3.0
```

**Comment by KristofferC** [comment]
Yes but you can change the code in the constructor to allow for that is what I mean.

**Comment by fredrikekre** [comment]
This works now btw:

``` jl
julia> A = rand(2,2); Asym = A + A.';

julia> Tensor{2,2}(Asym)
2x2 ContMechTensors.Tensor{2,2,Float64,4}:
 0.329017  1.10931 
 1.10931   0.989858

julia> SymmetricTensor{2,2}(Asym)
2x2 ContMechTensors.SymmetricTensor{2,2,Float64,3}:
 0.329017  1.10931 
 1.10931   0.989858
```

File: src/ContMechTensors.jl [hunk 2]
```diff
@@ -161,23 +161,23 @@ function call{dim}(Tt::Type{Vec{dim}}, data)
...
 end
 
-## These are some kinda ugly stuff to create different type of constructors.
+# These are some kinda ugly stuff to create different type of constructors.
 @gen_code function call{order, dim}(Tt::Union{Type{Tensor{order, dim}}, Type{SymmetricTensor{order, dim}}},
                                           data)
     # Check for valid orders
-    if !(order in (1,2,4))
-        @code (throw(ArgumentError("Only tensors of order 1, 2, 4 supported")))
+    if get_type(Tt) <: Tensor
```
Comment (KristofferC) [inline src/ContMechTensors.jl:168]: Should probably just split this up into two functions now, one for Tensors and one for SymmetricTensors. Can do it later.

**Pull request merged**